[broken Windows test][receiver/hostmetricsreceiver] TestScrapeMetrics_MuteErrorFlags/*muted

### Component(s)

_No response_

### Describe the issue you're reporting

This test is broken on Windows, it doesn't return 4 metrics as in the other OSes. Introduced via https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/58ee30b6bbe6c940690c63d5f8882fc5e7b2895b

https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6707698682/job/18227881486?pr=28682#step:6:804
```terminal
--- FAIL: TestScrapeMetrics_MuteErrorFlags (0.01s)
    --- FAIL: TestScrapeMetrics_MuteErrorFlags/Process_User_Error_Muted (0.00s)
        process_scraper_test.go:1059: 
            	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go:1059
            	Error:      	Not equal: 
            	            	expected: 4
            	            	actual  : 0
            	Test:       	TestScrapeMetrics_MuteErrorFlags/Process_User_Error_Muted
    --- FAIL: TestScrapeMetrics_MuteErrorFlags/Process_User_Error_Unmuted (0.00s)
        process_scraper_test.go:1059: 
            	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go:1059
            	Error:      	Not equal: 
            	            	expected: 4
            	            	actual  : 0
            	Test:       	TestScrapeMetrics_MuteErrorFlags/Process_User_Error_Unmuted
        process_scraper_test.go:1064: 
            	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go:1064
            	Error:      	An error is expected but got nil.
            	Test:       	TestScrapeMetrics_MuteErrorFlags/Process_User_Error_Unmuted
```

--- a/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go
+++ b/receiver/hostmetricsreceiver/internal/scraper/processscraper/process_scraper_test.go
@@ -935,6 +935,7 @@ func TestScrapeMetrics_MuteErrorFlags(t *testing.T) {
 
 	type testCase struct {
 		name                 string
+		skipTestCase         bool
 		muteProcessNameError bool
 		muteProcessExeError  bool
 		muteProcessIOError   bool
@@ -1000,6 +1001,7 @@ func TestScrapeMetrics_MuteErrorFlags(t *testing.T) {
 		},
 		{
 			name:                 "Process User Error Muted",
+			skipTestCase:         runtime.GOOS != "linux",
 			muteProcessUserError: true,
 			skipProcessNameError: true,
 			muteProcessExeError:  true,
@@ -1008,6 +1010,7 @@ func TestScrapeMetrics_MuteErrorFlags(t *testing.T) {
 		},
 		{
 			name:                 "Process User Error Unmuted",
+			skipTestCase:         runtime.GOOS != "linux",
 			muteProcessUserError: false,
 			skipProcessNameError: true,
 			muteProcessExeError:  true,
@@ -1019,6 +1022,9 @@ func TestScrapeMetrics_MuteErrorFlags(t *testing.T) {
 
 	for _, test := range testCases {
 		t.Run(test.name, func(t *testing.T) {
+			if test.skipTestCase {
+				t.Skipf("skipping test %v on %v", test.name, runtime.GOOS)
+			}
 			config := &Config{MetricsBuilderConfig: metadata.DefaultMetricsBuilderConfig()}
 			if !test.omitConfigField {
 				config.MuteProcessNameError = test.muteProcessNameError

